primitives/src/block.rs

```diff
@@ -374,4 +374,59 @@ mod tests {
         let version = Version::from_consensus(0b00100000000000000000000000000010);
         assert!(!Version::is_signalling_soft_fork(&version, 0));
     }
+
+    #[test]
+    fn version_to_consensus() {
+        let version = Version::from_consensus(1234567890);
+        assert_eq!(version.to_consensus(), 1234567890);
+    }
+
+    // Test the creation of a dummy block
+    #[cfg(feature = "alloc")]
+    #[test]
+    fn block_creation() {
+        use crate::locktime::absolute;
+        use crate::merkle_tree::TxMerkleNode;
+        use crate::{transaction, Amount, Sequence, Witness};
+        use crate::script::ScriptBuf;
+
+        let create_dummy_tx = |txid, version, lock_time, sequence| Transaction {
+            version,
+            lock_time: absolute::LockTime::from_consensus(lock_time),
+            input: vec![transaction::TxIn {
+                previous_output: transaction::OutPoint { txid, vout: 0 },
+                script_sig: ScriptBuf::new(),
+                sequence: Sequence(sequence),
+                witness: Witness::new(),
+            }],
+            output: vec![transaction::TxOut {
+                value: Amount::ONE_SAT,
+                script_pubkey: ScriptBuf::new(),
+            }],
+        };
+
+        let txid1 =
+            transaction::Txid::from_byte_array(hashes::sha256::Hash::hash(&[1]).to_byte_array());
+        let txid2 =
+            transaction::Txid::from_byte_array(hashes::sha256::Hash::hash(&[2]).to_byte_array());
+        let tx1 = create_dummy_tx(txid1, transaction::Version::ONE, 2, 1);
+        let tx2 = create_dummy_tx(txid2, transaction::Version::TWO, 3, 2);
+        let transactions = vec![tx1.clone(), tx2.clone()];
+
+        let header = Header {
+            version: Version::ONE,
+            prev_blockhash: BlockHash::from_byte_array([0x99; 32]),
+            merkle_root: TxMerkleNode::from_byte_array([0x77; 32]),
+            time: 2,
+            bits: CompactTarget::from_consensus(3),
+            nonce: 4,
+        };
+        let witness_root = Some(WitnessMerkleNode::from_byte_array([0x88; 32]));
+        let block = Block::new_unchecked(header, transactions.clone()).assume_checked(witness_root);
+
+        assert_eq!(block.transactions(), transactions);
+        assert_eq!(block.cached_witness_root(), witness_root);
+        assert_eq!(block.header(), &header);
+        assert_eq!(block.block_hash(), header.block_hash());
```

Comment:
Do I understand correctly that this test only checks getters?

Reply:
Yes, these were what were causing the problems with cargo mutants.

Reply:
I rekon we would be better off telling `cargo mutants`  to ignore getters. That would mean you could remove the whole `block_creation` test. I'm going to review it just for educational value though.

Also while playing with it I found a bug in `primitives` - so WIN for you.

```rust
    // Test the creation of a dummy block
    #[cfg(feature = "alloc")]
    #[test]
    fn block_creation() {
        use crate::{absolute, transaction, Amount, Sequence, ScriptBuf, TxMerkleNode, Witness};
```

This should work but `ScriptBuf` is not re-exported at `primitives` crate root. It should be because we want `bitcoin::Foo` to be imply `primitives::Foo` (and `units::Foo`) for all `Foo`.

Reply:
I opened #4018 to re-export `ScriptBuf` and a few others that were also missing